tests/test_TFEngine.py

```diff
@@ -1020,6 +1020,149 @@ def make_net_dict(task):
   engine.finalize()
 
 
+def check_train_and_search_two_targets(net_dict):
+  """
+  Tests training and search for network architectures having two targets ("classes_0", "classes_1")
+  and two corresponding output layers ("decision_0", "decision_1").
+  """
+  from MetaDataset import MetaDataset
+  from TFUtil import DimensionTag
+  from test_HDFDataset import generate_hdf_from_other
+
+  n_data_dim = 2
+  n_classes_dim_0 = 7
+  n_classes_dim_1 = 8
+
+  data_0 = {"class": "DummyDataset", "input_dim": n_data_dim, "output_dim": n_classes_dim_0,
+    "num_seqs": 2, "seq_len": 5}
+  data_0 = generate_hdf_from_other(data_0)
+  data_1 = {"class": "DummyDataset", "input_dim": n_data_dim, "output_dim": n_classes_dim_1,
+    "num_seqs": 2, "seq_len": 5}
+  data_1 = generate_hdf_from_other(data_1)
+
+  data = MetaDataset(datasets={"data_0": data_0, "data_1": data_1},
+    data_map={
+      "data": ("data_1", "data"),
+      "classes_0": ("data_0", "classes"),
+      "data_1": ("data_1", "data"),
+      "classes_1": ("data_1", "classes")},
+  )
+  data.init_seq_order()
+
+  dec_time = DimensionTag(kind=DimensionTag.Types.Spatial, description="dec time")
+
+  config = Config()
+  config.update({
+    "model": "%s/model" % _get_tmp_dir(),
+    "batch_size": 5000,
+    "max_seqs": 2,
+    "extern_data": {"data": {"dim": n_data_dim, "sparse": False},
```

Comment:
(Ah, works again)
This should be an issue on it's own. When using MetaDataset I have to mark the time dims as equal, also for in real configs. Is this expected?

Reply:
Yes, this is expected. How/why should it assume that they are the same?

Reply:
So the idea is that each data key has its own time dim unless configured otherwise, right? But it's quite common that multiple data keys are on the same time dim (at least for me ;) ). This should be easier to configure, without importing DimensionTag into the config.